File: Contests/Codeforces/Educational92DIV2/B.cpp
#include <bits/stdc++.h>
using namespace std;

#define ll long long
#define pb push_back
#define mp make_pair
#define ar array
#define ff first
#define ss second

int main() {
	ios_base::sync_with_stdio(0); cin.tie(0);

#ifndef ONLINE_JUDGE
  // for getting input from input.txt
  freopen("input.txt", "r+", stdin);
  // for writing output to output.txt
  freopen("output.txt", "w+", stdout);
#endif

    int t;
    cin >> t;
    while(t--) {
    	int n,k,z;
    	cin >> n >> k >> z;
    	int a[n];
    	for(int i=0;i<n;i++) {
    		cin >> a[i];
    	}
    	int ans = 0;
    	for(int i=0;i<=z;i++) {
    		int maxPair = -1;
    		int sum = 0;
    		int adjSum = 0;
    		for(int j=0;j<=k-2*i;j++) {
    			sum+=a[j];
    			adjSum = a[j] + a[j+1];
    			maxPair = max(maxPair,adjSum);
    		}
    		// cout << sum << '\n';
    		sum += i*maxPair;
    		ans = max(ans,sum);
    	}
    	cout << ans << '\n';
    }
}

File: Contests/Codeforces/Educational92DIV2/B.py
import sys 
# For getting input from input.txt file 
sys.stdin = open('input.txt', 'r')  
  
# Printing the Output to output.txt file 
sys.stdout = open('output.txt', 'w') 

for _ in range(int(input())):
	n, k, z = map(int, input().split())
	a = [int(x) for x in input().split()]
	ans = 0
	s = 0
	mx = 0
	for t in range(z + 1):
		pos = k - 2 * t
		if pos < 0:
			continue
		mx = 0
		s = 0
		for i in range(pos + 1):
			if i < n - 1:
				mx = max(mx, a[i] + a[i + 1])
			s += a[i]
			print(i)
		ans = max(ans, s + mx * t)
		print(t,s+mx*t,s,mx)
	print(ans)